### PR TITLE
refactor createAPIService

### PR DESCRIPTION
**Pull request by jiabaow**
- refactor API service builder
- refactor Lastfm method
- reformat code in all files (based on android studio suggestions) [sorry guys, I should have done this in a separate PR...]

### CODE REVIEW

File: app/src/main/java/ch/sdp/vibester/GenreTemporary.kt [hunk 1]
```diff
@@ -53,28 +50,29 @@ class GenreTemporary : AppCompatActivity() {
         startActivity(newIntent)
     }
 
+    // view is needed, remove it will result failure in tests
```
[the remark this answers is not in the document]
Comment (jiabaow) [reply app/src/main/java/ch/sdp/vibester/GenreTemporary.kt:52]: function call with different parameters, cannot really refactoring

File: app/src/androidTest/java/ch/sdp/vibester/activity/AuthenticationActivityTest.kt [hunk 2]
```diff
@@ -26,7 +26,7 @@ import kotlin.random.Random
  */
 @RunWith(AndroidJUnit4::class)
 class AuthenticationActivityTest {
-    private val SLEEP_TIME: Long = 5000
+    private val sleepTime: Long = 5000
```
Comment (zwierski) [inline app/src/androidTest/java/ch/sdp/vibester/activity/AuthenticationActivityTest.kt:29]: Should we use camel case for all the constants from now on?

**Review by zwierski — APPROVED** [review]
Good job on refactoring everything, the code feels much more legible! 👍

File: app/src/androidTest/java/ch/sdp/vibester/activity/AuthenticationActivityTest.kt (hunk 2 above)
Comment (jiabaow) [reply app/src/androidTest/java/ch/sdp/vibester/activity/AuthenticationActivityTest.kt:29]: I will say yes but only when they are inside a class

**Review by laurislopata — APPROVED** [review]
Don't have much to add. The code looks better after refactoring. LGTM

**Review by Tsathogguaa — APPROVED** [review]
Great work! Thanks for the refactoring!

**Pull request merged**